users/migrations/0006_user_global_id.py

```diff
@@ -0,0 +1,36 @@
+# Generated by Django 4.2.19 on 2025-03-11 12:53
+
+from django.db import migrations, models
+
+
+def transfer_global_id(apps, schema_editor):
+    """Transfer the social uid to the user global_id field"""
+    UserSocialAuth = apps.get_model("social_django", "UserSocialAuth")
+    User = apps.get_model("users", "User")
+
+    users_with_uid = User.objects.annotate(
+        social_uid=models.Subquery(
+            UserSocialAuth.objects.filter(
+                user=models.OuterRef("pk"),
+            ).values("uid")[:1]
+        )
+    )
+
+    users_with_uid.update(global_id=models.F("social_uid"))
```

Comment:
Should be careful here, this could be an issue with large numbers of users because it takes a row-level lock on the rows effected by this query, plus the whole migration is wrapped in a transaction. So this is effectively a full table lock which could either fail or cause errors on anything else trying to update the row (something as trivial as login which updates `last_login` could hit this). You probably want to do this in batches like I did here: https://github.com/mitodl/mit-learn/blob/main/users/migrations/0005_set_user_scim_id.py#L19

Note: it needs to be done in a separate migration that has `atomic = False` so that it doesn't wrap the migration in a transaction